scripts/complete_liquidity_provision.js

```diff
@@ -64,15 +67,24 @@ module.exports = async callback => {
 
     assert(argv.fleetSize % 2 == 0, "Fleet size must be a even number for easy deployment script")
 
-    console.log("1. Check for sufficient funds")
+    console.log("1. Sanity checks")
     if (!(await checkSufficiencyOfBalance(targetToken, masterSafe.address, investmentTargetToken))) {
-      callback(`Error: MasterSafe has insufficient balance for the token ${targetToken.address}.`)
+      return callback(`Error: MasterSafe has insufficient balance for the token ${targetToken.address}.`)
     }
     if (!(await checkSufficiencyOfBalance(stableToken, masterSafe.address, investmentStableToken))) {
-      callback(`Error: MasterSafe has insufficient balance for the token ${stableToken.address}.`)
+      return callback(`Error: MasterSafe has insufficient balance for the token ${stableToken.address}.`)
+    }
+    // check price against dex.ag's API
+    const targetTokenId = argv.targetToken
+    const stableTokenId = argv.stableToken
+    const priceCheck = await isPriceReasonable(exchange, targetTokenId, stableTokenId, argv.targetPrice, artifacts)
+    const proceed = proceedAnyways(priceCheck)
+
+    if (!priceCheck || !proceed) {
```

Comment:
This line makes the code fail if any one of `priceCheck` and `proceed` is false. But we want to ignore `priceCheck` if `proceed` is true.

Reply:
No... Shit this error corrects the upper one and that was why everything worked as expected